WebView/app/src/main/java/com/android/samples/webviewdemo/MainActivity.kt

```diff
@@ -59,6 +65,38 @@ class MainActivity : AppCompatActivity() {
 
     }
 
+    /** Instantiate the Listener  */
+    class MyListener() : WebViewCompat.WebMessageListener {
+        override fun onPostMessage(
+            view: WebView,
+            message: WebMessageCompat,
+            sourceOrigin: Uri,
+            isMainFrame: Boolean,
+            replyProxy: JavaScriptReplyProxy
+        ) {
+            replyProxy.postMessage("Got it!")
+            Log.i("grcoleman", "received onPostMessage in app")
+        }
+    }
+
+    /** Instantiate a class to determine which JS-Java API to use based on the version the application is running on */
+    class JsObject() {
```

Comment:
This doesn't need to be a separate class -- it can be directly on the `Activity`.

Reply:
Ok